src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+            guid = uuid.v4();
+            PreferencesManager.setViewState("GUID", guid);
+        }
+
+        oneTimeHealthData.guid = guid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.osVersion = HealthDataUtils.getOSVersion();
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (jsonData) {
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                })
+                .always(function () {
+                    PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime());
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if health data is to be send to the server. If user has enable the tracking, health data will be send for every 24 hours.
+     * If 24 hours or more than that has been passed, then send health data to the server
+     */
+    function _checkHealthDataExport() {
+        var result = new $.Deferred();
+        var isHDTracking = prefs.get("healthDataTracking");
+        window.clearTimeout(timeoutVar);
+
+        if (isHDTracking) {
+
+            var lastTimeSend = PreferencesManager.getViewState("lastTimeSendHealthData"),
+                currentTime = (new Date()).getTime();
+
+            if (!lastTimeSend) {
+                var randomTime = Math.floor(Math.random() * ONE_DAY);
+                lastTimeSend = currentTime + randomTime;
```

Comment:
@MarcelGerber lastTimeSend will be (currentTime + randomTime) only at first launch of Brackets when pref "var lastTimeSend = **PreferencesManager.getViewState("lastTimeSendHealthData")**" is not set. So for the first time it will be in the range of 24-48 hours. After that, it will be always 24 hour **timeoutVar = setTimeout(_checkHealthDataExport, ONE_DAY);**

Purpose of adding random time is to prevent enormous request to the server  It is possible as soon as new version of Brackets will be available users will update it and time of sending data to the server will possibly collide for many users. 

We need to have some minimum time period for sending data to the server which in our case is 24 hour.